### PR TITLE
docs: add sync api samples with json request

### PR DESCRIPTION
**Pull request by changsongd**
Previous PR is here: https://github.com/googleapis/python-optimization/pull/5

The previous PR was deleted because @dizcology suggested that using the auto-generated code samples can already serve our purpose and can also provide the benefits of easy maintainability. However, after reviewing our product documents, TPM and eng raised the concern that customers may feel confused about the linked code samples. The code sample construct the request in the code. But in reality, customers tend to construct the request in a json file. Thus they may find it hard to parse and use the json file without us showing them how to do it in code. This PR is to add a code snippet for the sync API that reads the request from a json file.

### CODE REVIEW

File: samples/snippets/sync_api.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  _, project_id = google.auth.default()
+  # Change the file name to your request file.
+  request_file_name = "resources/sync_request.json"
+  fleet_routing_client = optimization_v1.FleetRoutingClient()
+
+  with open(request_file_name, 'r') as f:
+    # The request must include the `parent` field with the value set to
+    # 'projects/{YOUR_GCP_PROJECT_ID}'.
+    fleet_routing_request = optimization_v1.OptimizeToursRequest.from_json(f.read())
+    fleet_routing_request.parent = "projects/{your_project_id}".format(your_project_id = project_id)
```
Comment (leahecole) [inline samples/snippets/sync_api.py:33]: Please use f-strings unless this is for a product supporting Python 2 or Python 3.5 or below

Comment (leahecole) [reply samples/snippets/sync_api.py:33]: Also, instead of having developers change their project ID here, follow the code snippet style guide where you have a TODO block elsewhere in the code for devs to fill in their values, and then use those values in your code https://g3doc.corp.google.com/company/teams/cloud-devrel/dpe/samples/snippets.md?cl=head

**Comment by leahecole** [comment]
I'm noticing that samples tests are passing but aren't actually running due to the lack of noxfile. I've flagged this to @googleapis/python-samples-owners as a concern to address. I'll send you individual instructions for running local samples tests as a workaround.

**Review by leahecole — CHANGES_REQUESTED** [review]
I added a few comments, but I also have a few overall requests
- Please add a `requirements.txt` file to the same directory your samples are in with libraries needed for your sample to run and have them pinned to specific versions (for example - https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/composer/rest/requirements.txt)
- Please add a `requirements.txt` file to the same directory your samples are in with libraries needed for tests to run. It will likely be identical to this one since your test doesn't call any additional 3rd party libraries (https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/composer/rest/requirements-test.txt)
- Please add a `noxfile-config.py` file to the directory where your samples are. This will enforce type hinting and specify which versions of Python to test in (by default, we test in 3.7, 3.8, 3.9, and 3.10. If your product does not support one of those versions, please exempt it in this file by adding the version to the list with 2.7 and 3.6). You can copy [this file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/noxfile_config.py) directly. This configuration file will interact with a [noxfile](https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/python_samples/noxfile.py.j2) that will be generated once a requirements file is present. The noxfile configures [nox](https://nox.thea.codes/en/stable/), our test runner
- Please add type hints, they are required per the authoring guide

**Review by xiliucity — APPROVED** [review]

**Comment by leahecole** [comment]
@changsongd I pinged you about why the noxfile reappeared! one that was generated SHOULD be there now - sorry - so confusing, I know!

**Comment by changsongd** [comment]
> @changsongd I pinged you about why the noxfile reappeared! one that was generated SHOULD be there now - sorry - so confusing, I know!

Oh, I think I accidentally deleted it. Will the OwlBot be triggered again and re-generate that file?

File: samples/snippets/sync_api.py (hunk 1 above)
Comment (changsongd) [reply samples/snippets/sync_api.py]: Updated to use f-strings. Also added the TODO comment and moved project_id to function input arg for easier testing.

File: noxfile.py [hunk 2]
```diff
@@ -303,3 +301,4 @@ def docfx(session):
         os.path.join("docs", ""),
         os.path.join("docs", "_build", "html", ""),
     )
+
```
Comment (leahecole) [inline noxfile.py:304]:
```suggestion
```

File: noxfile.py [hunk 3]
```diff
@@ -75,7 +75,6 @@
 @nox.session(python=DEFAULT_PYTHON_VERSION)
 def lint(session):
     """Run linters.
```
Comment (leahecole) [inline noxfile.py:77]:
```suggestion
    """Run linters.
    
```

File: noxfile.py [hunk 4]
```diff
@@ -75,7 +75,7 @@
...
-
+
```
Comment (leahecole) [inline noxfile.py:78]:
```suggestion
```

File: noxfile.py [hunk 5]
```diff
@@ -238,7 +237,6 @@ def system(session):
 @nox.session(python=DEFAULT_PYTHON_VERSION)
 def cover(session):
     """Run the final coverage report.
```
Comment (leahecole) [inline noxfile.py:239]:
```suggestion
    """Run the final coverage report.
    
```

File: noxfile.py [hunk 6]
```diff
@@ -238,7 +238,6 @@ def system(session):
...
```
Comment (leahecole) [inline noxfile.py:240]:
```suggestion
    """Run the final coverage report.

```

File: noxfile.py [hunk 7]
```diff
@@ -302,4 +302,4 @@ def docfx(session):
         os.path.join("docs", "_build", "doctrees", ""),
         os.path.join("docs", ""),
         os.path.join("docs", "_build", "html", ""),
-    )
+    )
```
Comment (leahecole) [inline noxfile.py:305]:
```suggestion
    ) 
```

File: noxfile.py [hunk 8]
```diff
@@ -302,4 +302,4 @@ def docfx(session):
...
```
Comment (leahecole) [inline noxfile.py:305]:
```suggestion
    )
```

**Comment by leahecole** [comment]
All of my noxfile commits are from resolving whitespace errors, in case anyone else is following along 😅

File: noxfile.py (hunk 7 above)
Comment (leahecole) [inline noxfile.py:305]:
```suggestion
    )
```

**Comment by leahecole** [comment]
I've now enabled the maps plugin and will retry

**Comment by leahecole** [comment]
Weird, I'm seeing 403s on two versions of Python but not on two others? 🤔 I'll try a re-run to see if it's consistent

**Comment by leahecole** [comment]
Still trying to get the samples lint check to show up

**Comment by leahecole** [comment]
trigger owlbot because it looks a lil stuck

**Review by leahecole — APPROVED** [review]
